src/sdl2/messagebox.rs

```diff
@@ -48,3 +101,60 @@ pub fn show_simple_message_box(flags: MessageBoxFlag, title: &str,
         Err(SdlError(get_error()))
     }
 }
+
+/// returns the id of the button triggered
+pub fn show_message_box<'a>(flags:MessageBoxFlag, buttons:&'a [ButtonData], title:&str,
+    message:&str, window:Option<&WindowRef>, scheme:Option<MessageBoxColorScheme>)
+    -> Result<i32,ShowMessageError> {
+    use self::ShowMessageError::*;
+    let mut button_id : c_int = 0;
+    let result = unsafe {
+        let title = match CString::new(title) {
```

Comment:
I don't think you need to allocate new strings here (unless messagebox is mutating these things, in which case wtf?) - Rust strings can be cast into happy happy *const c_chars for SDL2 to use.

Look here for an example:

https://github.com/AngryLawyer/rust-sdl2/blob/648e448cb718a34ce90c3e36a70083b1635fe89a/src/sdl2/hint.rs#L16